scripts/test_solidity_lsp.py

```diff
@@ -0,0 +1,379 @@
+            if test_fn.__doc__ != None:
+                title = test_fn.__doc__.strip()
+            print(f"{SGR_TEST_BEGIN}Testing {title} ...{SGR_RESET}")
+            try:
+                with JsonRpcProcess(self.solc_path, ["--lsp"]) as solc:
+                    test_fn(solc)
+                    self.tests_passed = self.tests_passed + 1
+            except ExpectationFailed as e:
+                print(f"{e}")
+                self.tests_failed = self.tests_failed + 1
+
+        print(
+            f"\nSummary:\n\n"
+            f"  Test cases: {self.tests_passed} passed, {self.tests_failed} failed\n"
+            f"  Assertions: {self.assertions_passed} passed, {self.assertions_failed} failed\n"
+        )
+
+        return min(self.assertions_failed, 127)
+
+    def setup_lsp(self, lsp: JsonRpcProcess, expose_project_root=True):
+        """
+        Prepares the solc LSP server by calling `initialize`,
+        and `initialized` methods.
+        """
+        project_root_uri = 'file://' + self.project_root_dir
+        params = {
+            'processId': None,
+            'rootPath': self.project_root_dir,
+            'rootUri': project_root_uri,
+            'trace': 'off',
+            'workspaceFolders': [
+                {'name': 'solidity-lsp', 'uri': project_root_uri}
+            ],
+            'initializationOptions': {},
+            'capabilities': {
+                'textDocument': {
+                    'publishDiagnostics': {'relatedInformation': True}
+                },
+                'workspace': {
+                    'applyEdit': True,
+                    'configuration': True,
+                    'didChangeConfiguration': {'dynamicRegistration': True},
+                    'workspaceEdit': {'documentChanges': True},
+                    'workspaceFolders': True
+                }
+            }
+        }
+        if expose_project_root == False:
+            params['rootUri'] = None
+            params['rootPath'] = None
+        lsp.call_method('initialize', params)
+        lsp.send_notification('initialized')
+
+    # {{{ helpers
+    def get_test_file_path(self, test_case_name):
+        return f"{self.project_root_dir}/{test_case_name}.sol"
+
+    def get_test_file_uri(self, test_case_name):
+        return "file://" + self.get_test_file_path(test_case_name)
+
+    def get_test_file_contents(self, test_case_name):
+        return open(self.get_test_file_path(test_case_name), mode="r", encoding="utf-8").read()
+
+    def extract_params_for_method(self, method_name: str, message: Any) -> Union[Any, None]:
+        if message['method'] != method_name:
+            return None
+        return message['params']
+
+    def open_file_and_wait_for_diagnostics(self,
+                                           solc: JsonRpcProcess,
+                                           test_case_name: str,
+                                           diagnostic_reports: int = 1) -> List[Any]:
```

Comment:
I think these names do not make it clear enough what you're dealing with (e.g. `diagnostic_reports` does not sound like an `int` to me, for moment I thought it was a list of reports when I saw it later in code).

Is `diagnostic_reports` meant to be the number of reports you expect? I see that the code will still accept a lower number.

```suggestion
                                           solc_process: JsonRpcProcess,
                                           test_case_name: str,
                                           max_diagnostic_reports: int = 1) -> List[Any]:
```

Reply:
i'm not sure about also renaming solc, of course i can, but where do we stop? There's a gray line to draw when it comes to readable names.

Reply:
We're using it for solc path in scripts so much that by now I'm conditioned to think about a path when I see it. And overall it's just ambiguous. `solc_process` is not overly long and you know what you're dealing with immediately when you see it.

Overall I think it's better to have longer names than cryptic or ambiguous ones. Longer to write but requires much less context to decode.